### PR TITLE
Vectorize Python loops with numpy operations

### PR DESCRIPTION
**Pull request by coleandreoli**
#774

### CODE REVIEW

**Comment by blankjul** [comment]
Thanks @coleandreoli for the contribution! We split this one.

**Merged (3 commits):**
- f2e06f3 `grad_jax.py` — import cleanup + vectorized basis matrix construction
- fcb3664 `misc.py` `repair()` → `np.clip` (identical semantics, cleaner)
- f8a0ad9 `pcx.py` — vectorized orthogonal direction loop; wrapped in `np.errstate` to suppress the transient NaN warning at the index parent slot

**Not merged — `IntegerRandomSampling`, `PermutationRandomSampling`, `random_permutations`:**

We ran benchmarks across a range of `n_var` sizes and found the picture is more nuanced than it first appears. For small `n_var` the vectorized approach is faster, but the crossover is around `n_var ≈ 400` — beyond that `np.argsort` is actually slower than numpy's internal Fisher-Yates shuffle, because sorting is O(n log n) vs O(n):

```
n_var      original    vectorized
   10        233 µs        32 µs   (new 7x faster)
  100        310 µs       174 µs   (new 1.8x faster)
  200        429 µs       355 µs   (new 1.2x faster)
  500       1087 µs      1319 µs   (orig 1.2x faster)
 1000       2013 µs      2750 µs   (orig 1.4x faster)
```

On top of that, the changes alter how the RNG state is consumed, so the same seed produces a different initial population and different tournament pairings — a silent reproducibility break for anyone relying on `seed=X` to reproduce results. pymoo is used a lot in research so we have to be careful there.

Really appreciate the effort though — the `pcx` and `repair` cleanups were great catches!
